File: backend/src/BidCalculationTool.Domain/Dto/FeeBreakdownItem.cs
namespace BidCalculationTool.Domain.Dto;

/// <summary>
/// Represents a single fee item in the breakdown of total costs.
/// This allows for dynamic fee structures without frontend coupling.
/// </summary>
public record FeeBreakdownItem
{
    /// <summary>
    /// Internal name/key of the fee (e.g., "BasicBuyerFee")
    /// </summary>
    public required string Name { get; init; }

    /// <summary>
    /// Human-readable display name (e.g., "Basic Buyer Fee")
    /// </summary>
    public required string DisplayName { get; init; }

    /// <summary>
    /// The calculated fee amount
    /// </summary>
    public required decimal Amount { get; init; }

    /// <summary>
    /// Optional description for tooltip or additional information
    /// </summary>
    public string? Description { get; init; }
}

File: frontend/src/modules/BidCalculator/dto/FeeBreakdownItem.ts
/**
 * Individual fee item in the dynamic fee breakdown.
 * This allows the frontend to render fees generically without being coupled to specific fee types.
 */
export interface FeeBreakdownItem {
  /** Internal name/key of the fee (e.g., "BasicBuyerFee") */
  name: string
  /** Human-readable display name (e.g., "Basic Buyer Fee") */
  displayName: string
  /** The calculated fee amount */
  amount: number
  /** Optional description for tooltip or additional information */
  description?: string
}

